src/clj/api/resolvers.clj

```diff
@@ -65,6 +65,42 @@
   [{:keys [db]} _ {bayes-factor-analysis-id :id :as parent}]
   (log/info "bayes-factor-analysis->bayes-factors" parent)
   (let [{:keys [bayes-factors]} (bayes-factor-model/get-bayes-factors db {:bayes-factor-analysis-id bayes-factor-analysis-id})
-        bayes-factors (json/read-str bayes-factors)]
+        bayes-factors           (json/read-str bayes-factors)]
     (log/info "bayes-factor-analysis->bayes-factors" {:bayes-factors bayes-factors})
     (clj->gql bayes-factors)))
+
+(defn search-user-analysis
+  "Returns paginated user analysis, following the Relay specification:
+  https://relay.dev/graphql/connections.htm.
+  Clients can ask for the next page using an opaque cursor."
+  [{:keys [db authed-user-id]} {first-n  :first
+                                after    :after
+                                statuses :statuses
+                                :or      {statuses [:UPLOADED
+                                                    :ATTRIBUTES_PARSED
+                                                    :ARGUMENTS_SET
+                                                    :QUEUED
+                                                    :RUNNING
+                                                    :SUCCEEDED
+                                                    :ERROR]}
+                                :as      args} _]
+  (log/info "search-user-analysis" args)
+  (let [after                 (if after
+                                (-> after decode-base64 Integer/parseInt inc)
+                                0)
+        {:keys [total-count]} (user-model/count-user-analysis db {:user-id authed-user-id :statuses (map name statuses)})
+        results               (user-model/search-user-analysis db {:user-id  authed-user-id
+                                                                   :statuses (map name statuses)
+                                                                   :limit    first-n
+                                                                   :offset   after})
+        edges                 (map-indexed (fn [index item] (hash-map :cursor (+ after index)
```

Comment:
Small detail but why the hash-map fn and not just a literal map here?

Reply:
no idea, fixed :man_shrugging: